pilot/pkg/networking/util/util.go

```diff
@@ -102,7 +102,7 @@ func BuildAddress(ip string, port uint32) core.Address {
 // TODO(mostrowski): consider passing map around to save iteration.
 func GetByAddress(listeners []*xdsapi.Listener, addr string) *xdsapi.Listener {
 	for _, listener := range listeners {
-		if listener.Address.String() == addr {
+		if listener != nil && listener.Address.String() == addr {
```

Comment:
shouldn't we be fixing the code that appends a nil object to the listener array?   seems like otherwise we could potentially run into this problem in other sections of code

Reply:
How do you test that? Simply check before appending? It seemed that none of the listeners we generate are nil, but somehow, the array/slice we create had a bunch of trailing nil elements. This was a hack to tackle such situations..

Reply:
will address a better solution in a follow-up PR